### ISSUE TITLE
Tests in co19_2 should be marked with a language version

### ISSUE THREAD

**Issue by leafpetersen** [dart-lang/co19#1082]
To simplify the testing framework, we are marking all of the pre Dart 2.12 language tests with a language marker.  This was done for all of the SDK tests [here](https://dart-review.googlesource.com/c/sdk/+/196548).  The co19_2 tests should follow on: that is, all tests in co19_2 which are not otherwise marked with a language version should be marked with `// @dart = 2.9` before the first directive of the test.  More discussion [here](https://github.com/dart-lang/sdk/issues/45577).

There is a tool in the Dart sdk tools directory called `tools/opt_files_out.dart` which can be used to automatically mark all files recursively under a given directory with `// @dart = 2.9`.  It will automatically skip any files which are already marked.  

Let me know if there are any concerns or questions.

cc @johnniwinther @eernstg @athomas @munificent

**Comment by iarkh** [comment]
`tools/opt_files_out.dart`  tool places `// @dart = 2.9` at the very beginning of the source file, i.e. before the `Copyright` comments section.

Is it correct or we should place it below the `Copyright` and `assertion` comments sections instead?

**Comment by athomas** [comment]
The reason for this is that co19 tests use license headers like this:
```dart
/*
 * Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
 * for details. All rights reserved. Use of this source code is governed by a
 * BSD-style license that can be found in the LICENSE file.
 */
```
rather than:
```dart
// Copyright (c) 2011, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
```

In general, block comments are discouraged in Dart (see for instance https://dart.dev/guides/language/effective-dart/documentation#dont-use-block-comments-for-documentation and https://dart.dev/guides/language/effective-dart/documentation#do-use--doc-comments-to-document-members-and-types).

So we want to clean this up as well (also in the master branch). Then the tool would work on the pre-nnbd (co19_2) branch.

**Comment by sgrekhov** [comment]
@leafpetersen @athomas @eernstg I've updated copyrights, changed test headers comments to be documentation-style and  added language version to all co19_2 tests. Please see https://github.com/dart-lang/co19/commit/ae517f5268215979d846697aa638f7801ceae072

But now, when I run these tests locally I get dozens of warnings like
```
2021-05-05 17:07:52.533972 Warning: /home/sgrekhov/Google/dart-sdk/sdk/tests/co19_2/src/LanguageFeatures/Instantiate-to-bound/typedef/static/typedef_l2_05_t19.dart: Invalid expectation '[typedef G<X extends Y' on line 46: /// with two related parameters: [typedef G<X extends Y, Y extends A<X>> = void
```
Line, mentioned in this particular warning warning is the following

```
/// @description Checks that instantiate-to-bounds works correctly for [typedef]
/// with two related parameters: [typedef G<X extends Y, Y extends A<X>> = void     <- line 46
/// Function<X1 extends X>()]
/// @author iarkh@unipro.ru
```
What are these warninhs?

**Comment by eernstg** [comment]
I'm afraid it's because the comment is a DartDoc comment (`///`) rather than a regular comment (`//`). This implies that certain tools will try to interpret parts of the contents of the comment, and that probably didn't occur previously.

I suspect that there is no need for any DartDoc comments in co19 tests at all, in which case it might be possible to run a very simple script (a single `sed` command, perhaps) that changes `///` at the beginning of the line to `//`.

**Comment by leafpetersen** [comment]
That's odd, it looks like maybe the test harness is interpreting that line as a static error expectation?  cc @munificent is this a bug in the expectation parser?

**Comment by munificent** [comment]
It's a "feature" in the *multitest* parser. :-/

When multitests were first created, the syntax for defining a multitest section was `///`. Not too long after, we started using `///` for doc comments which obviously made multitests pretty confusing. To avoid that, we added support for `//#` as multitest markers in addition to `///`. Almost all multitests use `//#` now.

A while back, I went on a mission to remove support for the old `///` syntax. I fixed the uses I could find in our tests and then filed #391 here for the two cases I found in co19. Now that that's fixed (thanks!), I can probably just remove support for `///` from the multitest parser and these warnings will go away.

However, when I search for `///` in co19 today, I see a lot of lines like:

```
   sendPort.send(", " + i); /// static type warning
```

(This is co19_2/src/LibTest/isolate/Isolate/spawnUri_A05_t01_isolate.dart.) Do you know what those are about? They *look* sort of like multitest markers, but aren't because there's no `:` or section name after the `///`.

Either way, here's a patch that removes support for the old `///` multitest markers: https://dart-review.googlesource.com/c/sdk/+/198445

Let me know if that doesn't fix the warnings. Thanks for all the many years of maintaining Dart tests. :)

**Comment by sgrekhov** [comment]
> (This is co19_2/src/LibTest/isolate/Isolate/spawnUri_A05_t01_isolate.dart.) Do you know what those are about? They look sort of like multitest markers, but aren't because there's no : or section name after the ///.

It's part of old style expectation of compile-errors and static warnings. When expected result of a test is `@static-warning` then we also add `/// static type warning` to exact place where we expect a warning. I don't think that it is actualy used by infra, it's more for the information. Anyway there is an issue https://github.com/dart-lang/co19/issues/1086 requesting to remove all old-style `@compile-error` and `@static-warning` expectations. So these `/// static type warning` will be removed

Thank you!

**Comment by sgrekhov** [comment]
Fixed by 4d36e3a 1556227 57bc7ac

**Comment by sgrekhov** [comment]
@eernstg thank you for pointing! Removed excessive copyrights
